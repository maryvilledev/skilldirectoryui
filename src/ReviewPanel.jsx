import React, {PropTypes} from 'react'
import { Panel, Col } from 'react-bootstrap'

const ReviewPanel = ({review}) => {
  if(!review) return <div />;

  const body=review.body;
  const time = review.timestamp;
  const good = review.positive;
  const reviewer = review.team_member_name;
  const header = reviewer + " on " + new Date(time).toDateString();
  let color = null;
  if (good) {
    color = "success"
  }
  else {
    color = "danger"
  }

  return (
  <div>
    <Panel header={header} bsStyle={color} >
      <Col>{body}</Col>
    </Panel>
  </div>
  );
}

ReviewPanel.propTypes = {
  review: PropTypes.object.isRequired
}

export default ReviewPanel;
