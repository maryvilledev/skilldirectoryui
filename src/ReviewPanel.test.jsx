import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ReviewPanel from './ReviewPanel';

describe('<ReviewPanel />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ReviewPanel />, div);
  });

  it('matches the stored snapshot', () => {
    const wrapper = shallow(<ReviewPanel />);
    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
