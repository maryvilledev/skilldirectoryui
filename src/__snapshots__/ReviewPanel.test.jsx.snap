exports[`<ReviewPanel /> matches the stored snapshot 1`] = `<div />`;
